src/components/SingleAssetEntryExit.svelte

```diff
@@ -0,0 +1,39 @@
+<script>
+    import images from '../config/images.json'
+    import smartPools from '../config/pools.json'
+    import Modal from '../components/elements/Modal.svelte';
+    import Swap from '../components/JoinSwapExternAmount.svelte';
+    import { pools } from '../stores/eth.js'
+
+    export let buyTokenAddress;
+
+    // Bind to parent function to create a custom activator
+    export let openTrigger = () => swapModal.open();
+
+    let swapModal;
+
+    $: listed = $pools[buyTokenAddress.toLowerCase()];
+    $: symbol = smartPools[buyTokenAddress.toLowerCase()].symbol;    
+    
+</script>
+<section>
+    <slot>
+        <img src={images.defi_pp_merge} alt="defi single asset exchange"
+            class="hidden md:block cursor-pointer"
+            on:click={() => { openTrigger() }}
```

Comment:
I am a fan of just passing the references: on:click={openTrigger} but this is a matter of personal preference

Reply:
Certainly looks better